phoenix-core/src/main/java/org/apache/phoenix/execute/LiteralResultIterationPlan.java

```diff
@@ -85,7 +85,8 @@ public void close() throws SQLException {
 
             @Override
             public Tuple next() throws SQLException {
-                while (!this.closed && (offset != null && count++ < offset) && tupleIterator.hasNext()) {
+                while (!this.closed && (offset != null && count < offset) && tupleIterator.hasNext()) {
+                    count++;
```

Comment:
What I meant was this "count++" doesn't seem to work with LIMIT(below) together. LIMIT should be number of rows returned starting from the OFFSET row, right?